### PR TITLE
Add shellcheck and fix warnings

### LINKED ISSUE #1670: Fix shell scripts warnings
### Context
Similier to `rust clippy` command, there's a static analysis tool for shell scripts too.  [shellcheck ](https://github.com/koalaman/shellcheck). It's simple to use  `shellcheck xxx.sh`. More further, we can add it to `Makefile`, for example
```Makefile
check-scripts:
    # Fail if any of these files have warnings
    shellcheck scripts/*.sh
```

shell scripts contains in 
1. scripts
2. tee-work/scripts

Here's some warnings and suggestions from scripts:

```shell

Did you mean: 
$PARACHAIN_BIN --alice --collator --force-authoring --tmp --chain "$CHAIN"-dev \


In launch-local-binary.sh line 105:
  --bootnodes /ip4/127.0.0.1/tcp/30336/p2p/$RELAY_ALICE_IDENTITY &> "para.alice.log" &
                                           ^-------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --bootnodes /ip4/127.0.0.1/tcp/30336/p2p/"$RELAY_ALICE_IDENTITY" &> "para.alice.log" &


In launch-local-binary.sh line 109:
cd "$ROOTDIR/ts-tests"
^--------------------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean: 
cd "$ROOTDIR/ts-tests" || exit


In launch-local-binary.sh line 118:
cd "$ROOTDIR/ts-tests"
^--------------------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean: 
cd "$ROOTDIR/ts-tests" || exit
```

### PR DESCRIPTION
**Pull request by Kailai-Wang**
resolves #1670 

It should be straightforward. The severity level is set to `warning` to avoid spam.

### CODE REVIEW

**Comment by kziemianek** [comment]
Can we run these checks as a part of `scripts/pre-commit.sh` ?

**Comment by Kailai-Wang** [comment]
> Can we run these checks as a part of `scripts/pre-commit.sh` ?

I could add it - it's not enforced by CI tho, as shell script is just tooling script, it either works or fails. Normally we are fine if it does the job.

Edit: done.

File: scripts/generate-upstream-patch.sh [hunk 1]
```diff
@@ -52,7 +53,8 @@ generate_upstream_patch() {
 	echo "fetch upstream_$TARGET"
 	git fetch -q "upstream_$TARGET"
 
-	local tmp_dir=$(mktemp -d)
+	local tmp_dir
+	tmp_dir=$(mktemp -d)
```
Comment (arieljbirnbaum) [inline scripts/generate-upstream-patch.sh:57]: Really? That's an odd convention imho but if it makes the bot happy then okay :D

File: scripts/geth/run_geth.sh [hunk 2]
```diff
@@ -57,7 +57,7 @@ start="${GETH_BIN} --datadir ${DATADIR} \
     --miner.etherbase 0xff93B45308FD417dF303D6515aB04D9e89a750Ca
 "
 if [ ${DOCKER} = "local" ]; then
-    eval ${start} 2>&1 &>'geth.log' &
+    eval ${start} &>'geth.log' &
```
Comment (arieljbirnbaum) [inline scripts/geth/run_geth.sh:60]: Just to make sure; if we're running locally then we want the output stream to go into `geth.log` and the error stream to go into the terminal?

File: scripts/generate-upstream-patch.sh (hunk 1 above)
Comment (Kailai-Wang) [reply scripts/generate-upstream-patch.sh:57]: Yea: https://www.shellcheck.net/wiki/SC2155

File: scripts/geth/run_geth.sh (hunk 2 above)
Comment (BillyWooo) [reply scripts/geth/run_geth.sh:60]: Similar question: why remove it?

Comment (Kailai-Wang) [reply scripts/geth/run_geth.sh:60]: doesn't `&>` redirect both stdout and stderr to `geth.log`? It supersedes the previous redirection. So `2>&1` is redundant here

Comment (BillyWooo) [reply scripts/geth/run_geth.sh:60]: Yes. You are right. Merged into one step is enough.

**Review by BillyWooo — APPROVED** [review]

**Pull request merged**